### PR TITLE
Use build agent's cert store when running e2e tests

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Ensures that node uses the certificate store of the build agent when running e2e tests. Without this, it defaults to using its bundled certs, so it doesn't "see" the self-signed cert that we added to the system cert store in a previous step, and our r11s e2e tests all time out because of that (long story).

## Reviewer Guidance

It seems our e2e tests pipeline [doesn't like `test/` branches](https://dev.azure.com/fluidframework/internal/_build/results?buildId=106915&view=logs&j=5d40b943-5a8e-529c-ed4b-95198832abfc&t=fb8cf0ae-41a8-58f6-d156-329de70d0970) (line 39), so I can't fully validate that the tests now execute successfully without first investigating and addressing the apparent issue that `test-version-utils` seems to have with `test/` branches. But the invocation seems correct:

image

The e2e tests for r11s are almost all timing out right now so this can't make things much worse. The new flag will get passed to e2e tests against all targets (local, tinylicious, r11s, frs, odsp), but I don't expect it to have any impact on the other targets.

### CODE REVIEW

File: tools/pipelines/templates/include-test-real-service.yml [hunk 1]
```diff
@@ -270,7 +270,7 @@ jobs:
         inputs:
           command: 'custom'
           workingDir: ${{ parameters.testWorkspace }}/node_modules/${{ parameters.testPackage }}
-          customCommand: 'run ${{ parameters.testCommand }} -- ${{ variant.flags }}'
+          customCommand: 'run ${{ parameters.testCommand }} -- ${{ variant.flags }} --use-openssl-ca'
```
Comment (Josmithr) [inline tools/pipelines/templates/include-test-real-service.yml:273]: Might be worth adding a brief comment here explaining why we need the flag, for future reference.

**Review by Josmithr — APPROVED** [review]
Changes look good based on your explanation in the PR description.

File: tools/pipelines/templates/include-test-real-service.yml (hunk 1 above)
Comment (alexvy86) [reply tools/pipelines/templates/include-test-real-service.yml:273]: Fair, adding it now. I expect it won't be necessary anymore once we're back on public certs, but doesn't hurt in the meantime.

**Pull request merged**